### PR TITLE
[vault] reauthenticate on missing data if not authenticated

### PR DESCRIPTION
**Pull request by maorfr**
following #1348, we now know what secrets are forbidden.

those secrets are usually not really forbidden, and my suspicion is that we are hitting the SA token limit. this makes sense since we recently moved to a singleton vault client, which means we are using the token without refreshing it.

this PR adds a refresh every time we find a secret forbidden and we find that the client is not authenticated.

### CODE REVIEW

**Comment by arilivigni** [comment]
LGTM

File: reconcile/utils/vault.py [hunk 1]
```diff
@@ -160,10 +164,19 @@ def read(self, secret):
         secret_field = secret['field']
         secret_format = secret.get('format', 'plain')
         secret_version = secret.get('version')
-        try:
+
+        kv_version = self._get_mount_version_by_secret_path(secret_path)
+
+        data = None
+        if kv_version == 2:
```
Comment (apahim) [inline reconcile/utils/vault.py:171]: Are you sure that read_v2 won't raise an exception when trying to read from a v1 engine? I believe you verified that... just checking anyway :)

**Review by jfchevrette — APPROVED** [review]

File: reconcile/utils/vault.py (hunk 1 above)
Comment (maorfr) [reply reconcile/utils/vault.py:171]: with this change we are no longer trying to read v1 from v2 or vice versa

Comment (maorfr) [reply reconcile/utils/vault.py:171]: same as here: https://github.com/app-sre/qontract-reconcile/pull/1414/files#diff-421d17fb36e3edd0412b07a396bcdb5a518bd98d1abfc44d5df8f6072d589540R86-R92

**Comment by apahim** [comment]
lgtm

**Pull request merged**